### PR TITLE
A11y

### PR DESCRIPTION
**Pull request by Hypercubed**
## Summary

* Add cypress axe
* Add button focus outlines
* Use Sematic HTML in inputs, selects
* Add aruia roles to plus menu

## Checklist

- [x] \*Added unit tests
- [x] \*Added a code reviewer
- [x] Added changes to `/projects/swimlane/ngx-ui/CHANGELOG.md` under HEAD (Unreleased)
- [x] Updated the demo page
- [x] Included screenshots of visual changes

_\*required_

image

### CODE REVIEW

File: cypress/integration/forms/buttons.spec.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    beforeEach(() => {
+      cy.get('[sectiontitle="Buttons"]').as('SUT');
+    });
+
+    it('has no detectable a11y violations on load', () => {
+      cy.get('@SUT').within(() => {
+        cy.get('.ngx-section-content button').then($el => {
+          cy.checkA11y($el, {
+            rules: {
+              'color-contrast': { enabled: false }
```
Comment (barbados-clemens) [inline cypress/integration/forms/buttons.spec.js:18]: question: this is waiting on design to come through with updated designs with appropriate color designs right? if so probably should leave a TODO to make sure we enable the contrast check after those designs are in.

File: cypress/integration/forms/buttons.spec.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        cy.get('@consoleLog').should('be.calledWith', 'Demo app click: Default');
+
+        cy.realPress('Tab'); // Navigates to first button
+        cy.focused().should('have.text', 'Primary').findPseudoAfter().its('opacity').should('eq', '1');
+        cy.realPress('Space'); // Presses the button
+        cy.get('@consoleLog').should('be.calledWith', 'Demo app click: Primary');
+      });
+    });
+
+    it('does not fucus on disable buttons', () => {
```
Comment (barbados-clemens) [inline cypress/integration/forms/buttons.spec.js:57]: nitpick: typo 
```suggestion
    it('does not focus on disable buttons', () => {
```

File: cypress/integration/forms/buttons.spec.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    beforeEach(() => {
+      cy.get('[sectiontitle="ngx-button"]').as('SUT');
+    });
+
+    it('has no detectable a11y violations on load', () => {
+      cy.get('@SUT').within(() => {
+        cy.get('.ngx-section-content ngx-button').withinEach($el => {
+          cy.checkA11y($el, {
+            rules: {
+              'color-contrast': { enabled: false }
```
Comment (barbados-clemens) [inline cypress/integration/forms/buttons.spec.js:83]: same question about leaving TODO for future enablement

File: cypress/integration/forms/selects.spec.ts [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
     cy.visit('/selects');
+    cy.injectAxe();
     cy.get('.page-loader').should('not.exist', { timeout: 20000 });
   });
 
+  it('have no detectable a11y violations on load', () => {
+    cy.get('ngx-select-input').withinEach($el => {
+      cy.checkA11y($el, {
+        rules: {
+          'color-contrast': { enabled: false },
```
Comment (barbados-clemens) [inline cypress/integration/forms/selects.spec.ts:12]: same color contrast question about leaving future TODO

File: projects/swimlane/ngx-ui/src/lib/components/button/button.component.scss [hunk 5]
```diff
@@ -3,10 +3,41 @@
 
 .ngx-button {
   transition: background-color 0.25s ease-out !important;
+  padding: 0 !important;
```
Comment (barbados-clemens) [inline projects/swimlane/ngx-ui/src/lib/components/button/button.component.scss:6]: suggestion: anyway we can not use `!important` here?

File: projects/swimlane/ngx-ui/src/lib/components/input/input.component.ts [hunk 6]
```diff
@@ -250,7 +250,7 @@ export class InputComponent implements AfterViewInit, OnDestroy, ControlValueAcc
   private _required = false;
   private _autoSelect = false;
   private _autofocus = false;
-  private _autocomplete = false;
+  private _autocomplete = 'off';
```
Comment (barbados-clemens) [inline projects/swimlane/ngx-ui/src/lib/components/input/input.component.ts:253]: nitpick: adding type `'on' | 'off' | boolean` for stronger typing
```suggestion
  private _autocomplete: 'on' | 'off' = 'off';
```

**Review by barbados-clemens — COMMENTED** [review]
style suggestion and a couple of questions. 

Side note about Focus Ring. you mentioned that outline/border isn't really that "ideal" I came across this blog post by Discord who build a custom component to make a "Focus Ring" around an arbitrary group of elements. they did this in React, but wondering if we might be able to do the same but with directives 🤔. Just a thought!
https://blog.discord.com/how-discord-implemented-app-wide-keyboard-navigation-abf073fd71de

File: projects/swimlane/ngx-ui/src/lib/components/input/input.component.ts (hunk 6 above)
Comment (Hypercubed) [reply projects/swimlane/ngx-ui/src/lib/components/input/input.component.ts:253]: Going with `boolean | string`

**Review by barbados-clemens — APPROVED** [review]

**Pull request merged**